superset/utils/core.py

```diff
@@ -148,6 +148,10 @@ class GenericDataType(IntEnum):
     STRING = 1
     TEMPORAL = 2
     BOOLEAN = 3
+    ARRAY = 4
+    JSON = 5
+    MAP = 6
+    ROW = 7
```

Comment:
We need to make sure this logic is in synced with `superset-ui/core` - for now it might be a good idea to just map all complex types to `STRING` as has previously been done, and later introduce more complex generic types once we add proper support for them.